js/sites.js

```diff
@@ -259,35 +285,22 @@
      */
     captureIfMissing() {
       if (!document.hidden && !this.link.imageURI) {
-        gNewTab.sendToBrowser("NewTab:BackgroundPageThumbs", {
-          url: this.url
+        gNewTab.sendToBrowser("NewTab:CaptureBackgroundPageThumbs", {
+          link: this.link,
         });
       }
     },
 
     /**
-     * Refreshes the thumbnail for the site.
-     */
-    refreshThumbnail() {
-      gNewTab.sendToBrowser("NewTab:PageThumbs", {
-        link: this.link
-      });
-    },
-
-    /**
-     * Render the correct thumbnail for the site.
-     *
-     * @param {Object} aData Contains enhanced links and the URI generated from the
-     *        current URL.
+     * Show the enhanced thumbnail.
      */
-    getURI(aData) {
+    showEnhancedThumbnail() {
       let thumbnail = this._querySelector(".newtab-thumbnail");
       if (this.link.bgColor) {
         thumbnail.style.backgroundColor = this.link.bgColor;
       }
 
-      let uri = this.link.imageURI || aData.uri;
-      thumbnail.style.backgroundImage = `url("${uri}")`;
+      thumbnail.style.background = `url("${this.link.imageURI}")`;
```

Comment:
I spoke to @emtwo about this, it is the implementation of the old `enhanced tile` feature.

`enhanced` tiles used to mean that a history tile could become `enhanced` if it had an `enhancedImageURI`.

But this also happens to be the only place where the `directoryId` is set, so the functionality needs to be preserved for reporting purposes.

What we can do is to set the `directoryId` if the tile is of type `ENHANCED` and write a note to set the type to `enhanced` if it was a history tile later on, when `DirectoryLinksProvider` gets implemented in the content process.